### ISSUE TITLE
Break out Core Metadata Service

### ISSUE THREAD

**Issue by ckoppelman** [financialforcedev/apex-mdapi#73]
I would love to include some of the service in Apex scripts I run on orgs that I do not personally own.  In order to do that, I pare down MetadataService.cls, just keeping the features that I need, and include it at the top of my script.  In order to make it easier to do this, it would be terrific if this package were architected in a way that added on various metadata as "modules" instead of one monolithic class.

**Comment by afawcett** [comment]
:+1: Great idea and use case, it is huge and only getting bigger with every release! I would need some 'metadata' of its own to list the types related to each module. What kind of modules would you suggest?

**Comment by ckoppelman** [comment]
The easiest breakpoint is pulling out all of the non-operational code from the Service (you seem to begin all the operational classes with lowercase letters, like `checkDeployStatus_element`).   Then pull out testing code (e.g., CodeCoverageWarning).  I don't have the time right now for careful analysis, but I assume there are other natural breakpoints of self-contained modules, like Flow,  ServiceCloud, Dashboard/Reports, Territory, Users/Roles/Permissions, etc.  I'd use the Salesforce Setup menu as a reasonable guide since your user base ought to be familiar with that, and it's mostly logical (probably makes sense to squish together good chunks of standard objects, though).

**Comment by wes1278** [comment]
I would love to see not only this, but perhaps we can just pare down a branch/tag of this repo to only include the files that we need. I use the fantastic github deploy tool that @afawcett wrote to deploy this. and I really don't want all the pages/tabs/examples deployed to every org I want this wrapper in. I only want the service class (and perhaps my new proposed remotesitepage).

Any ideas?

**Comment by afawcett** [comment]
Yes good idea, make this a core repo and then create an examples one. Agree.

**Comment by afawcett** [comment]
I've just created the new [repo](https://github.com/financialforcedev/apex-mdapi-examples) if you fancied helping out with the split... :+1:

**Comment by wes1278** [comment]
Want to make me a contributor of that new repo so I don't have another fork? I've got most of it separated already.

**Comment by afawcett** [comment]
That's great really appreciate it. Though I'm not in a position to add you as a contributor right now sorry.
